test/integration/model/findAll/group.test.js

```diff
@@ -51,6 +51,49 @@ describe(Support.getTestDialectTeaser('Model'), () => {
           expect(parseInt(posts[1].get('comment_count'))).to.be.equal(2);
         });
       });
+
+      it('should not add primary key when grouping using a belongsTo association', () => {
+        const Post = current.define('Post', {
+          id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
+          name: { type: DataTypes.STRING, allowNull: false }
+        });
+
+        const Comment = current.define('Comment', {
+          id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
+          text: { type: DataTypes.STRING, allowNull: false }
+        });
+
+        Post.hasMany(Comment);
+        Comment.belongsTo(Post);
+
+        return current.sync({ force: true }).then(() => {
+          return Post.bulkCreate([
+            { name: 'post-1' },
+            { name: 'post-2' }
+          ]);
+        }).then(() => {
+          return Comment.bulkCreate([
+            { text: 'Market', PostId: 1 },
+            { text: 'Text', PostId: 2 },
+            { text: 'Abc', PostId: 2 },
+            { text: 'Semaphor', PostId: 1 },
+            { text: 'Text', PostId: 1 }
+          ]);
+        }).then(() => {
+          return Comment.findAll({
+            attributes: ['PostId', [Sequelize.fn('COUNT', Sequelize.col('Comment.id')), 'comment_count']],
+            include: [
+              { model: Post, attributes: [] }
+            ],
+            group: ['PostId']
+          });
+        }).then(posts => {
+          expect(posts[0].get().hasOwnProperty('id')).to.not.be.ok;
```

Comment:
`to.equal(false)` - I know this style is already used a lot of places in the codebase, but it's prone to uncaught typos